
From: Zinx Verituse <zinx@epicsol.org>

Posting again.  Still not applied as of 2.6.7.  The hid-tmff driver is
useless without this.

Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/drivers/usb/input/hid-tmff.c |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)

diff -puN drivers/usb/input/hid-tmff.c~hid-tmff-fix-again drivers/usb/input/hid-tmff.c
--- 25/drivers/usb/input/hid-tmff.c~hid-tmff-fix-again	Wed Jun 16 15:41:51 2004
+++ 25-akpm/drivers/usb/input/hid-tmff.c	Wed Jun 16 15:41:51 2004
@@ -110,7 +110,7 @@ int hid_tmff_init(struct hid_device *hid
 {
 	struct tmff_device *private;
 	struct list_head *pos;
-	struct hid_input *hidinput = list_entry(&hid->inputs, struct hid_input, list);
+	struct hid_input *hidinput = list_entry(hid->inputs.next, struct hid_input, list);
 
 	private = kmalloc(sizeof(struct tmff_device), GFP_KERNEL);
 	if (!private)
_
